source/source.go

```diff
@@ -74,6 +74,10 @@ type Service interface {
 	// Config defines a function that captures
 	// the pipeline configuration from a repo.
 	Config(user *library.User, org, name, ref string) ([]byte, error)
+	// ConfigBackoff is a truncated constant backoff wrapper.
+	// Calls Config to check if yml/yaml is valid.
+	// Retries every five seconds and returns an error after five tries.
+	ConfigBackoff(user *library.User, org, name, ref string) ([]byte, error)
```

Comment:
Similar to my previous comment, I think we'll want to update the description to be a bit more accurate 👍